pkg/services/extsvcauth/oauthserver/oasimpl/service.go

```diff
@@ -458,7 +458,12 @@ func (*OAuth2ServiceImpl) handleRegistrationPermissions(registration *extsvcauth
 
 // handlePluginStateChanged reset the client authorized grant_types according to the plugin state
 func (s *OAuth2ServiceImpl) handlePluginStateChanged(ctx context.Context, event *pluginsettings.PluginStateChangedEvent) error {
-	s.logger.Info("Plugin state changed", "pluginId", event.PluginId, "enabled", event.Enabled)
+	s.logger.Debug("Plugin state changed", "pluginId", event.PluginId, "enabled", event.Enabled)
+
+	if event.OrgId != extsvcauth.TmpOrgID {
+		s.logger.Debug("External Service not tied to this organization", "OrgId", event.OrgId)
+		return nil
+	}
```

Comment:
Is this check now needed because `handlePluginStateChanged` is a global (as in cross-org) plugin action?

Reply:
`handlePluginStateChanged` can be triggered by enabling the plugin on any organization yes :+1:

Reply:
But at least in a first release we won't support multiple organizations :grimacing: 
(Also note that this code is for the OAuth2 Server, which we might remove altogether)